compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticHintCode.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.util.diagnostic;
+
+import io.ballerina.tools.diagnostics.DiagnosticCode;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+/**
+ * This class contains a list of diagnostic hint codes.
+ *
+ * @since Swan Lake
+ */
+public enum DiagnosticHintCode implements DiagnosticCode {
+
+    UNNECESSARY_CONDITION("BCE40000", "unnecessary.condition");
```

Comment:
@dulajdilshan What does BCE stand for? If it's Ballerina Compiler Error, should we consider a different prefix for other categories of diagnostics?

Reply:
Yes. The earlier discussions were only to separate errors/warning/hints from the compiler side and the side of the extensions (having a different prefix for extensions). However, I'm +1 for having a different prefix for hints. (Ex: `BCH`). Also, this can be followed for warning too.

Reply:
Shall we raise this in Slack and check with others as well?

Reply:
@dulajdilshan 

> Wasn't there existing negative tests for this particular error? 🤔

Yes. there are.. :) i will fix them.. thats why pr fails